src/admin/localService/loadFile.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+const fs = require("fs");
+const pathJoin = require("path").join;
+
+const readdir = fs.promises.readdir;
+
+/**
+ * load file into server memory
+ */
+async function loadFile(credDir: string, fileName: string): Promise<string> {
+  const filePath = await getFilePath(`${credDir}/`, fileName);
+  if (!filePath)
+    return Promise.reject(
+      new Error(`${fileName} not found. Please enter a full cred repo`)
+    );
+
+  return fs.promises.readFile(pathJoin(filePath, fileName), "utf8");
+}
+
+/**
+ * server utility function used to naively locate ancillary files needed in the frontend:
+ * weightedGraph.json
+ * project.json
+ * pluginDeclarations.json
+ */
+async function getFilePath(dir: string, targetFile: string): Promise<string> {
+  const files: Dirent[] = await ((readdir(dir, {
+    withFileTypes: true,
+  }): any): Promise<Dirent[]>); // Dirent is not currently in the builtin node flow-typed lib
+  if (files.map((f) => f.name).includes(targetFile)) return dir;
+  const search = files
+    .filter((f) => f.isDirectory())
+    .map(async (f) => {
+      return await getFilePath(pathJoin(dir, f.name), targetFile);
+    });
+  if (search.length > 0) {
+    const results: Array<string> = await Promise.all(search);
+    const result = results.find((e) => !!e) || "";
+    return result;
+  }
+  return "";
+}
+
+type Dirent = {|
+  name: string,
+  type: number,
```

Comment:
Is this field used for anything? Noticed the `withFileTypes: true` option as well, but dont see anywhere that we check for it.

Reply:
This would not be strictly necessary, except that this is the canonical shape of Node's `Dirent` class. The reason I pass in the `withFileTypes` flag is so that we have access to the `isDirectory method: https://github.com/topocount/sourcecred/blob/1e37a2714933e8c3ac0b5659250f2d07e85fa05c/src/admin/localService/loadFile.js#L35

Otherwise a simple string with the name of file or directory is returned. Since I'm not even technically declaring a class, I can remove it to avoid any confusion, but it's included in Node Dirents, so I figured I should keep it in there.